tests/test_dem.py

```diff
@@ -1,47 +1,57 @@
 import pyposeidon.dem as pdem
+import xarray as xr
 import numpy as np
 import pytest
 
 from . import DATA_DIR
 
+source = xr.open_dataset(DATA_DIR / "dem.nc")
+s1 = source.sel(longitude=slice(-180, 0))
+s2 = source.sel(longitude=slice(0, 180))
+s1 = s1.assign_coords({"longitude": s1.longitude + 360})
+snew = xr.merge([s2, s1])
 
 DEM_SOURCES = pytest.mark.parametrize(
     "dem_source",
     [
         pytest.param(DATA_DIR / "dem.nc", id="local netcdf"),
+        pytest.param(snew, id="local netcdf remapped"),
         pytest.param(DATA_DIR / "dem.tif", id="local geotiff"),
     ],
 )
 
 WINDOWS = pytest.mark.parametrize(
-    "kwargs",
+    "kwargs, expected",
     [
         # fmt: off
-        pytest.param({"lon_min": -30, "lon_max": -10.0, "lat_min": 60.0, "lat_max": 70.0}, id="local 1"),
-        pytest.param({"lon_min": 175.0, "lon_max": 184.0, "lat_min": 14.5, "lat_max": 16.5}, id="local 2"),
-        pytest.param({"lon_min": 175.0 - 360.0, "lon_max": 184.0 - 360.0, "lat_min": 14.5, "lat_max": 16.5}, id="local 3"),
-        pytest.param({"lon_min": -180.0, "lon_max": 180.0, "lat_min": -90.0, "lat_max": 90.0}, id="global 2"),
-        pytest.param({"lon_min": -360.0, "lon_max": 0, "lat_min": -90.0, "lat_max": 90.0}, id="global 1"),
+        pytest.param({"lon_min": -30, "lon_max": -10.0, "lat_min": 60.0, "lat_max": 70.0}, (37,71), id="local 1"),
+        pytest.param({"lon_min": 175.0, "lon_max": 184.0, "lat_min": 14.5, "lat_max": 16.5}, (11,34), id="local 2"),
+        pytest.param({"lon_min": 175.0 - 360.0, "lon_max": 184.0 - 360.0, "lat_min": 14.5, "lat_max": 16.5}, (11,34), id="local 3"),
+        pytest.param({"lon_min": -10., "lon_max": 5., "lat_min": 46., "lat_max": 65.}, (68,54), id="local 4"),
+        pytest.param({"lon_min": -180.0, "lon_max": 180.0, "lat_min": -90.0, "lat_max": 90.0}, (600,1200), id="global 2"),
+        pytest.param({"lon_min": -360.0, "lon_max": 0, "lat_min": -90.0, "lat_max": 90.0}, (600,1200), id="global 1"),
         # fmt: on
     ],
 )
 
 
 @DEM_SOURCES
 @WINDOWS
-def test_dem(dem_source, kwargs):
+def test_dem(dem_source, kwargs, expected):
```

Comment:
Shouldn't we use `expected` in this function?

Reply:
The one function tests the shape of the output and the other the existence of NaNs.